components/brave_today/browser/publishers_controller.cc

```diff
@@ -113,6 +124,19 @@ void PublishersController::GetOrFetchPublishers(base::OnceClosure callback,
   EnsurePublishersIsUpdating();
 }
 
+void PublishersController::GetLocale(
+    mojom::BraveNewsController::GetLocaleCallback callback) {
+  if (publishers_.empty() || is_update_in_progress_) {
```

Comment:
nit: there is already a pattern for this that we don't have to copy here - basically just `GetOrFetchPublishers([](){ std::move(callback).Run(default_locale_); })`

`PublishersController::GetOrFetchPublishers(base::OnceClosure callback,
                                                bool wait_for_current_update)` is provided as a private member function for exactly this reason - do an operation once we have some data.

If you agree, we can fix as follow-up in the next PR.

Reply:
Sweet, just changed it over :smile: